internal/app/todo_app.go

```diff
@@ -0,0 +1,314 @@
+			initialDate = initialDate.AddDate(1, 0, 0)
+			if initialDate.After(now) {
+				return initialDate.Format("20060102"), nil
+			}
+		}
+
+	default:
+		return "", errors.New("unsupported repeat rule")
+	}
+}
+
+type Task struct {
+	ID      string `json:"id"`
+	Date    string `json:"date"`
+	Title   string `json:"title"`
+	Comment string `json:"comment,omitempty"`
+	Repeat  string `json:"repeat,omitempty"`
+}
+
+func AddTask(task *Task, db *sql.DB) (string, error) {
+	if task.Title == "" {
+		return "", errors.New("task title is required")
+	}
+
+	now := time.Now()
+	currentDay := now.Format("20060102")
+
+	if task.Date == "" || task.Date == "today" {
+		task.Date = currentDay
+	}
+
+	if _, err := time.Parse("20060102", task.Date); err != nil {
+		return "", errors.New("invalid date format, expected YYYYMMDD")
+	}
+
+	if task.Date < currentDay {
+		if task.Repeat != "" {
+			nextDate, err := NextDate(now, task.Date, task.Repeat)
+			if err != nil {
+				return "", fmt.Errorf("invalid repeat rule: %v", err)
+			}
+			task.Date = nextDate
+		} else {
+			task.Date = currentDay
+		}
+	}
+
+	if task.Repeat != "" {
+		validPrefixes := []string{"d", "y"}
+		isValid := false
+		for _, prefix := range validPrefixes {
+			if strings.HasPrefix(task.Repeat, prefix) {
+				isValid = true
+				break
+			}
+		}
+		if !isValid {
+			return "", errors.New("unsupported repeat rule")
+		}
+	}
+
+	var exists bool
+	checkQuery := `SELECT EXISTS(SELECT 1 FROM scheduler WHERE date = ? AND title = ?)`
+	if err := db.QueryRow(checkQuery, task.Date, task.Title).Scan(&exists); err != nil {
+		return "", fmt.Errorf("failed to check existing task: %v", err)
+	}
+	if exists {
+		return "", errors.New("task with the same date and title already exists")
+	}
+
+	query := `INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)`
+	res, err := db.Exec(query, task.Date, task.Title, task.Comment, task.Repeat)
+	if err != nil {
+		return "", fmt.Errorf("failed to save task: %v", err)
+	}
+
+	id, err := res.LastInsertId()
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve task ID: %v", err)
+	}
+
+	return fmt.Sprintf("%d", id), nil
+}
+
+func GetTasks(db *sql.DB) ([]Task, error) {
+	tasks := make([]Task, 0)
+
+	rows, err := db.Query(`SELECT id, date, title, comment, repeat FROM scheduler ORDER BY date LIMIT 50`)
+	if err != nil {
+		return nil, fmt.Errorf("failed to select tasks: %v", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var (
+			t    Task
+			id64 int64
+		)
+		if err := rows.Scan(&id64, &t.Date, &t.Title, &t.Comment, &t.Repeat); err != nil {
+			return nil, fmt.Errorf("failed to scan row: %v", err)
+		}
+		t.ID = strconv.FormatInt(id64, 10)
+		tasks = append(tasks, t)
+	}
+
+	return tasks, nil
+}
+
+func GetTaskByID(db *sql.DB, id string) (*Task, error) {
+	if id == "" {
+		return nil, errors.New("no task id provided")
+	}
+	idInt, err := strconv.Atoi(id)
+	if err != nil {
+		return nil, fmt.Errorf("invalid task id '%s'", id)
```

Comment:
Очень хорошо, что везде оборачиваешь, но лучше текст ошибок оставлять
```suggestion
		return nil, fmt.Errorf("invalid task id '%s': %w", id, err)
```